### PR TITLE
chore(swingset): add docs/images on comms garbage collection details

### PR DESCRIPTION
**Pull request by warner**
There's still a lot of TODOs and text to write, but this covers a lot of the comms GC details.

Use https://github.com/Agoric/agoric-sdk/blob/improve-gc-docs/packages/SwingSet/docs/garbage-collection.md to read the rendered form.

### CODE REVIEW

File: packages/SwingSet/docs/garbage-collection.md [hunk 1]
```diff
@@ -519,3 +474,121 @@ The delivery of each GC action is processed as follows:
...
+
+The comms vat does not use real JavaScript objects: there are no Remotables, Presences, Representatives, or Promises, which makes the code somewhat simpler than liveslots. However it must deal with asynchronous connections to the remotes, which makes it more complex than the kernel. When it sends a `dropExport` over the wire, that message won't be delivered immediately, and it might receive an arbitrary number of messages that were sent before the `dropExport` arrives. This complicates the reachability tracking.
```
Comment (FUDCo) [inline packages/SwingSet/docs/garbage-collection.md]: "more complex than the kernel" -- this aspect is more complex than the kernel, but the kernel as a whole is considerably more complex than the comms vat, no?

File: packages/SwingSet/docs/garbage-collection.md [hunk 2]
```diff
@@ -519,3 +474,121 @@ The delivery of each GC action is processed as follows:
...
+
+## Comms Tables
+
+The comms vat has an object table (key-value entries prefixed with `lo$NN`) that tracks an `.owner` for each object: the remote which exported that object (or empty to mean it came from the kernel). It also maintains `.reachable` and a `.recognizable` refcount for each. As with the kernel, these counts are only maintained for the downstream importers, not for the exporter. (Promises also have a refcount, but this only tracks usage by resolved promise data, not remotes: promises are retired when they are resolved, not when importers drop them).
+
+The comms vat has a c-list for each remote, plus one for the kernel. This c-list table has a special "isReachable" flag for each object. On the importer, this describes the state of the importer: `true` means they can reach (and recognize) the object, `false` means they can merely recognize it. On the exporter, this tracks whether we've send a `drop` or not.
```
Comment (FUDCo) [inline packages/SwingSet/docs/garbage-collection.md]: "On the exporter, this tracks whether we've send a `drop` or not" -- this is confusing.  Presumably `true` means reachable means have not sent a `drop`, which is inverted from what a reader might infer from reading that.

File: packages/SwingSet/docs/garbage-collection.md [hunk 3]
```diff
@@ -519,3 +474,121 @@ The delivery of each GC action is processed as follows:
...
+
+The upstream comms vat will send `remote.retireImport` messages to its downstream importing systems. This will eventually cause the final importing comms vat to do a `syscall.retireExport` (since that comms vat was exporting the object into the kernel), which eventually makes that kernel do `dispatch.retireImport` into the vat. This will allow the downstream vat to delete the WeakMap/WeakSet entry, releasing any data used as a value.
+
+The sequence of downstream messages looks like this:
+
+![Comms Retire Sequence (downstream) diagram](./images/gc/comms-retire-sequence-downstream.png "Comms Retire Sequence (downstream)")
+
+## Informed vs Ignorant, Seqnum and Acknum
+
+Despite our best efforts, the speed of light is not infinite, and messages sent from one comms vat to another will not arrive immediately. Machine A might send a message to machine B about some object which B is trying to drop, and these two messages might "cross on the wire". We use sequence and acknowledgment numbers to keep track of who knew what, and when, so we can properly handle the race between these crossing messages.
```
Comment (FUDCo) [inline packages/SwingSet/docs/garbage-collection.md:544]: We should try to do something about the speed of light once we have the resources.

**Review by FUDCo — APPROVED** [review]
Looks good so far, modulo some minor phrasing infelicities that I commented on.

File: packages/SwingSet/docs/garbage-collection.md (hunk 1 above)
Comment (warner) [reply packages/SwingSet/docs/garbage-collection.md]: replaced with "which makes the comms GC protocol more complex than that of the kernel"

File: packages/SwingSet/docs/garbage-collection.md (hunk 2 above)
Comment (warner) [reply packages/SwingSet/docs/garbage-collection.md]: changed to "On the exporter, it tracks whether we've send a `drop` or not: the flag is set to `true` until we send a `drop`, at which point it is cleared to `false`."

**Pull request merged**